### DISCUSSION TITLE
Corrupted versioning?

### DISCUSSION THREAD

**Discussion by sslaia** [AnySoftKeyboard/AnySoftKeyboard#3342] [Q&A]
Hi @menny I notice that every time I compile the language pack, the versionName becomes 4.1.-315 (with minus before the third number, 1 instead the usual 0 of the second number). And that what appears in addons/.generated_pack_version file. Is this correct? How to correct it if it's wrong? I didn't change anything in the ASK code. I just cloned the repository. Thanks.

**Comment by sslaia** [comment]
Any one else has had similiar issue? I cloned the repository again today and the issue persists. The repository from last year doesn't have this issue. So with the latest code, I can't compile new version of language addons any more.

**Reply by menny** [reply]
this should not happen, can you paste the gradle log while building?

**Reply by sslaia** [reply]
Is the gradle log the one under build/build-logging? The content is:

Build log created at 2022-05-11T20:40:02.179441Z
Build started at 2022-05-11T20:40:10.494952Z
Gradle 7.4.2
Build finished at 2022-05-11T20:40:22.156003Z
build action: Build
build result SUCCESS
build tasks:
* :addons:languages:nias:apk:preBuild state: SKIPPED
* :addons:languages:nias:apk:preDebugBuild state: SKIPPED
* :addons:languages:nias:apk:mergeDebugNativeDebugMetadata state: SKIPPED
* :api:preBuild state: SKIPPED
* :api:preDebugBuild state: SKIPPED
* :api:compileDebugAidl state: SKIPPED
* :api:compileDebugRenderscript state: SKIPPED
* :api:generateDebugBuildConfig state: SKIPPED
* :api:generateDebugResValues state: SKIPPED
* :api:generateDebugResources state: SKIPPED
* :api:packageDebugResources state: SKIPPED
* :api:parseDebugLocalResources state: EXECUTED
* :api:processDebugManifest state: EXECUTED
* :api:generateDebugRFile state: EXECUTED
* :api:javaPreCompileDebug state: SKIPPED
* :api:compileDebugJavaWithJavac state: EXECUTED
* :api:bundleLibCompileToJarDebug state: SKIPPED
* :addons:base:preBuild state: SKIPPED
* :addons:base:preDebugBuild state: SKIPPED
* :addons:base:compileDebugAidl state: SKIPPED
* :api:packageDebugRenderscript state: SKIPPED
* :addons:base:compileDebugRenderscript state: SKIPPED
* :addons:base:generateDebugBuildConfig state: SKIPPED
* :addons:base:generateDebugResValues state: SKIPPED
* :addons:base:generateDebugResources state: SKIPPED
* :addons:base:packageDebugResources state: SKIPPED
* :addons:base:parseDebugLocalResources state: EXECUTED
* :addons:base:processDebugManifest state: EXECUTED
* :addons:base:generateDebugRFile state: EXECUTED
* :addons:base:javaPreCompileDebug state: SKIPPED
* :addons:base:compileDebugJavaWithJavac state: EXECUTED
* :addons:base:bundleLibCompileToJarDebug state: SKIPPED
* :addons:languages:nias:pack:mergeAllWordLists state: SKIPPED
* :addons:languages:nias:pack:makeDictionary state: EXECUTED
* :addons:languages:nias:pack:preBuild state: EXECUTED
* :addons:languages:nias:pack:preDebugBuild state: EXECUTED
* :addons:languages:nias:pack:compileDebugAidl state: SKIPPED
* :addons:languages:nias:apk:compileDebugAidl state: SKIPPED
* :addons:base:packageDebugRenderscript state: SKIPPED
* :addons:languages:nias:pack:packageDebugRenderscript state: SKIPPED
* :addons:languages:nias:apk:compileDebugRenderscript state: SKIPPED
* :addons:languages:nias:apk:generateDebugBuildConfig state: SKIPPED
* :addons:languages:nias:apk:javaPreCompileDebug state: SKIPPED
* :api:compileDebugLibraryResources state: SKIPPED
* :addons:base:compileDebugLibraryResources state: SKIPPED
* :api:writeDebugAarMetadata state: SKIPPED
* :addons:base:writeDebugAarMetadata state: SKIPPED
* :addons:languages:nias:pack:writeDebugAarMetadata state: SKIPPED
* :addons:languages:nias:apk:checkDebugAarMetadata state: EXECUTED
* :addons:languages:nias:apk:generateDebugResValues state: SKIPPED
* :addons:languages:nias:apk:generateDebugResources state: SKIPPED
* :addons:languages:nias:pack:compileDebugRenderscript state: SKIPPED
* :addons:languages:nias:pack:generateDebugResValues state: SKIPPED
* :addons:languages:nias:pack:generateDebugResources state: SKIPPED
* :addons:languages:nias:pack:packageDebugResources state: SKIPPED
* :addons:languages:nias:apk:mergeDebugResources state: SKIPPED
* :addons:languages:nias:apk:createDebugCompatibleScreenManifests state: SKIPPED
* :api:extractDeepLinksDebug state: SKIPPED
* :addons:base:extractDeepLinksDebug state: SKIPPED
* :addons:languages:nias:apk:extractDeepLinksDebug state: SKIPPED
* :addons:languages:nias:pack:extractDeepLinksDebug state: SKIPPED
* :addons:languages:nias:pack:processDebugManifest state: EXECUTED
* :addons:languages:nias:apk:processDebugMainManifest state: EXECUTED
* :addons:languages:nias:apk:processDebugManifest state: EXECUTED
* :addons:languages:nias:apk:processDebugManifestForPackage state: EXECUTED
* :addons:languages:nias:pack:compileDebugLibraryResources state: SKIPPED
* :addons:languages:nias:pack:parseDebugLocalResources state: EXECUTED
* :addons:languages:nias:pack:generateDebugRFile state: EXECUTED
* :addons:languages:nias:apk:processDebugResources state: EXECUTED
* :addons:languages:nias:pack:generateDebugBuildConfig state: SKIPPED
* :addons:languages:nias:pack:javaPreCompileDebug state: SKIPPED
* :addons:languages:nias:pack:compileDebugJavaWithJavac state: EXECUTED
* :addons:languages:nias:pack:bundleLibCompileToJarDebug state: SKIPPED
* :addons:languages:nias:apk:compileDebugJavaWithJavac state: EXECUTED
* :api:mergeDebugShaders state: SKIPPED
* :api:compileDebugShaders state: SKIPPED
* :api:generateDebugAssets state: SKIPPED
* :api:packageDebugAssets state: SKIPPED
* :addons:base:mergeDebugShaders state: SKIPPED
* :addons:base:compileDebugShaders state: SKIPPED
* :addons:base:generateDebugAssets state: SKIPPED
* :addons:base:packageDebugAssets state: SKIPPED
* :addons:languages:nias:apk:mergeDebugShaders state: SKIPPED
* :addons:languages:nias:apk:compileDebugShaders state: SKIPPED
* :addons:languages:nias:apk:generateDebugAssets state: SKIPPED
* :addons:languages:nias:pack:mergeDebugShaders state: SKIPPED
* :addons:languages:nias:pack:compileDebugShaders state: SKIPPED
* :addons:languages:nias:pack:generateDebugAssets state: SKIPPED
* :addons:languages:nias:pack:packageDebugAssets state: SKIPPED
* :addons:languages:nias:apk:mergeDebugAssets state: SKIPPED
* :addons:languages:nias:apk:compressDebugAssets state: SKIPPED
* :api:processDebugJavaRes state: SKIPPED
* :api:bundleLibResDebug state: SKIPPED
* :addons:base:processDebugJavaRes state: SKIPPED
* :addons:base:bundleLibResDebug state: SKIPPED
* :addons:languages:nias:apk:processDebugJavaRes state: SKIPPED
* :addons:languages:nias:pack:processDebugJavaRes state: SKIPPED
* :addons:languages:nias:pack:bundleLibResDebug state: SKIPPED
* :addons:languages:nias:apk:mergeDebugJavaResource state: EXECUTED
* :api:bundleLibRuntimeToJarDebug state: SKIPPED
* :addons:base:bundleLibRuntimeToJarDebug state: SKIPPED
* :addons:languages:nias:pack:bundleLibRuntimeToJarDebug state: SKIPPED
* :addons:languages:nias:apk:checkDebugDuplicateClasses state: SKIPPED
* :addons:languages:nias:apk:dexBuilderDebug state: EXECUTED
* :addons:languages:nias:apk:desugarDebugFileDependencies state: EXECUTED
* :addons:languages:nias:apk:mergeExtDexDebug state: SKIPPED
* :addons:languages:nias:apk:mergeDexDebug state: SKIPPED
* :api:mergeDebugJniLibFolders state: SKIPPED
* :api:mergeDebugNativeLibs state: SKIPPED
* :api:copyDebugJniLibsProjectOnly state: SKIPPED
* :addons:base:mergeDebugJniLibFolders state: SKIPPED
* :addons:base:mergeDebugNativeLibs state: SKIPPED
* :addons:base:copyDebugJniLibsProjectOnly state: SKIPPED
* :addons:languages:nias:apk:mergeDebugJniLibFolders state: SKIPPED
* :addons:languages:nias:pack:mergeDebugJniLibFolders state: SKIPPED
* :addons:languages:nias:pack:mergeDebugNativeLibs state: SKIPPED
* :addons:languages:nias:pack:copyDebugJniLibsProjectOnly state: SKIPPED
* :addons:languages:nias:apk:mergeDebugNativeLibs state: SKIPPED
* :addons:languages:nias:apk:stripDebugDebugSymbols state: SKIPPED
* :addons:languages:nias:apk:validateSigningDebug state: SKIPPED
* :addons:languages:nias:apk:writeDebugAppMetadata state: SKIPPED
* :addons:languages:nias:apk:writeDebugSigningConfigVersions state: SKIPPED
* :addons:languages:nias:apk:packageDebug state: EXECUTED
* :addons:languages:nias:apk:createDebugApkListingFileRedirect state: EXECUTED
* :addons:languages:nias:apk:assembleDebug state: EXECUTED
* :api:preReleaseBuild state: SKIPPED
* :api:compileReleaseAidl state: SKIPPED
* :addons:base:preReleaseBuild state: SKIPPED
* :addons:base:compileReleaseAidl state: SKIPPED
* :addons:languages:nias:pack:preReleaseBuild state: EXECUTED
* :addons:languages:nias:pack:compileReleaseAidl state: SKIPPED
* :addons:languages:nias:pack:mergeReleaseJniLibFolders state: SKIPPED
* :addons:languages:nias:pack:mergeReleaseNativeLibs state: SKIPPED
* :addons:languages:nias:pack:stripReleaseDebugSymbols state: SKIPPED
* :addons:languages:nias:pack:copyReleaseJniLibsProjectAndLocalJars state: SKIPPED
* :api:compileReleaseRenderscript state: SKIPPED
* :api:generateReleaseBuildConfig state: SKIPPED
* :api:generateReleaseResValues state: SKIPPED
* :api:generateReleaseResources state: SKIPPED
* :api:packageReleaseResources state: SKIPPED
* :api:parseReleaseLocalResources state: SKIPPED
* :api:processReleaseManifest state: EXECUTED
* :api:generateReleaseRFile state: EXECUTED
* :api:javaPreCompileRelease state: SKIPPED
* :api:compileReleaseJavaWithJavac state: EXECUTED
* :api:bundleLibCompileToJarRelease state: SKIPPED
* :api:packageReleaseRenderscript state: SKIPPED
* :addons:base:compileReleaseRenderscript state: SKIPPED
* :addons:base:generateReleaseBuildConfig state: SKIPPED
* :addons:base:generateReleaseResValues state: SKIPPED
* :addons:base:generateReleaseResources state: SKIPPED
* :addons:base:packageReleaseResources state: SKIPPED
* :addons:base:parseReleaseLocalResources state: SKIPPED
* :addons:base:processReleaseManifest state: EXECUTED
* :addons:base:generateReleaseRFile state: EXECUTED
* :addons:base:javaPreCompileRelease state: SKIPPED
* :addons:base:compileReleaseJavaWithJavac state: EXECUTED
* :addons:base:bundleLibCompileToJarRelease state: SKIPPED
* :addons:base:packageReleaseRenderscript state: SKIPPED
* :addons:languages:nias:pack:compileReleaseRenderscript state: SKIPPED
* :addons:languages:nias:pack:generateReleaseBuildConfig state: SKIPPED
* :addons:languages:nias:pack:generateReleaseResValues state: SKIPPED
* :addons:languages:nias:pack:generateReleaseResources state: SKIPPED
* :addons:languages:nias:pack:packageReleaseResources state: SKIPPED
* :addons:languages:nias:pack:parseReleaseLocalResources state: SKIPPED
* :addons:languages:nias:pack:processReleaseManifest state: EXECUTED
* :addons:languages:nias:pack:generateReleaseRFile state: EXECUTED
* :addons:languages:nias:pack:extractReleaseAnnotations state: EXECUTED
* :addons:languages:nias:pack:javaPreCompileRelease state: SKIPPED
* :addons:languages:nias:pack:compileReleaseJavaWithJavac state: EXECUTED
* :addons:languages:nias:pack:mergeReleaseGeneratedProguardFiles state: SKIPPED
* :addons:languages:nias:pack:mergeReleaseConsumerProguardFiles state: SKIPPED
* :addons:languages:nias:pack:mergeReleaseShaders state: SKIPPED
* :addons:languages:nias:pack:compileReleaseShaders state: SKIPPED
* :addons:languages:nias:pack:generateReleaseAssets state: SKIPPED
* :addons:languages:nias:pack:packageReleaseAssets state: SKIPPED
* :addons:languages:nias:pack:packageReleaseRenderscript state: SKIPPED
* :addons:languages:nias:pack:prepareLintJarForPublish state: EXECUTED
* :addons:languages:nias:pack:prepareReleaseArtProfile state: EXECUTED
* :addons:languages:nias:pack:processReleaseJavaRes state: SKIPPED
* :addons:languages:nias:pack:mergeReleaseJavaResource state: EXECUTED
* :addons:languages:nias:pack:syncReleaseLibJars state: SKIPPED
* :addons:languages:nias:pack:writeReleaseAarMetadata state: SKIPPED
* :addons:languages:nias:pack:bundleReleaseLocalLintAar state: EXECUTED
* :addons:base:mergeReleaseJniLibFolders state: SKIPPED
* :addons:base:mergeReleaseNativeLibs state: SKIPPED
* :addons:base:stripReleaseDebugSymbols state: SKIPPED
* :addons:base:copyReleaseJniLibsProjectAndLocalJars state: SKIPPED
* :addons:base:extractReleaseAnnotations state: EXECUTED
* :addons:base:mergeReleaseGeneratedProguardFiles state: SKIPPED
* :addons:base:mergeReleaseConsumerProguardFiles state: SKIPPED
* :addons:base:mergeReleaseShaders state: SKIPPED
* :addons:base:compileReleaseShaders state: SKIPPED
* :addons:base:generateReleaseAssets state: SKIPPED
* :addons:base:packageReleaseAssets state: SKIPPED
* :addons:base:prepareLintJarForPublish state: EXECUTED
* :addons:base:prepareReleaseArtProfile state: EXECUTED
* :addons:base:processReleaseJavaRes state: SKIPPED
* :addons:base:mergeReleaseJavaResource state: EXECUTED
* :addons:base:syncReleaseLibJars state: SKIPPED
* :addons:base:writeReleaseAarMetadata state: SKIPPED
* :addons:base:bundleReleaseLocalLintAar state: EXECUTED
* :api:mergeReleaseJniLibFolders state: SKIPPED
* :api:mergeReleaseNativeLibs state: SKIPPED
* :api:stripReleaseDebugSymbols state: SKIPPED
* :api:copyReleaseJniLibsProjectAndLocalJars state: SKIPPED
* :api:extractReleaseAnnotations state: EXECUTED
* :api:mergeReleaseGeneratedProguardFiles state: SKIPPED
* :api:mergeReleaseConsumerProguardFiles state: SKIPPED
* :api:mergeReleaseShaders state: SKIPPED
* :api:compileReleaseShaders state: SKIPPED
* :api:generateReleaseAssets state: SKIPPED
* :api:packageReleaseAssets state: SKIPPED
* :api:prepareLintJarForPublish state: EXECUTED
* :api:prepareReleaseArtProfile state: EXECUTED
* :api:processReleaseJavaRes state: SKIPPED
* :api:mergeReleaseJavaResource state: EXECUTED
* :api:syncReleaseLibJars state: SKIPPED
* :api:writeReleaseAarMetadata state: SKIPPED
* :api:bundleReleaseLocalLintAar state: EXECUTED
* :api:bundleLibResRelease state: SKIPPED
* :api:bundleLibRuntimeToJarRelease state: SKIPPED
* :api:createFullJarRelease state: SKIPPED
* :api:writeReleaseLintModelMetadata state: SKIPPED
* :addons:base:bundleLibResRelease state: SKIPPED
* :addons:base:bundleLibRuntimeToJarRelease state: SKIPPED
* :addons:base:createFullJarRelease state: SKIPPED
* :addons:base:writeReleaseLintModelMetadata state: SKIPPED
* :addons:languages:nias:apk:preReleaseBuild state: SKIPPED
* :addons:languages:nias:apk:compileReleaseAidl state: SKIPPED
* :addons:languages:nias:apk:compileReleaseRenderscript state: SKIPPED
* :addons:languages:nias:apk:generateReleaseBuildConfig state: SKIPPED
* :addons:languages:nias:apk:javaPreCompileRelease state: SKIPPED
* :addons:languages:nias:apk:writeReleaseApplicationId state: EXECUTED
* :addons:languages:nias:apk:analyticsRecordingRelease state: EXECUTED
* :api:compileReleaseLibraryResources state: SKIPPED
* :addons:base:compileReleaseLibraryResources state: SKIPPED
* :addons:languages:nias:apk:checkReleaseAarMetadata state: SKIPPED
* :addons:languages:nias:apk:generateReleaseResValues state: SKIPPED
* :addons:languages:nias:apk:generateReleaseResources state: SKIPPED
* :addons:languages:nias:apk:mergeReleaseResources state: SKIPPED
* :addons:languages:nias:apk:createReleaseCompatibleScreenManifests state: SKIPPED
* :api:extractDeepLinksRelease state: SKIPPED
* :addons:base:extractDeepLinksRelease state: SKIPPED
* :addons:languages:nias:apk:extractDeepLinksRelease state: SKIPPED
* :addons:languages:nias:pack:extractDeepLinksRelease state: SKIPPED
* :addons:languages:nias:apk:processReleaseMainManifest state: EXECUTED
* :addons:languages:nias:apk:processReleaseManifest state: EXECUTED
* :addons:languages:nias:apk:processReleaseManifestForPackage state: EXECUTED
* :addons:languages:nias:pack:compileReleaseLibraryResources state: SKIPPED
* :addons:languages:nias:apk:processReleaseResources state: EXECUTED
* :addons:languages:nias:pack:bundleLibCompileToJarRelease state: SKIPPED
* :addons:languages:nias:apk:compileReleaseJavaWithJavac state: EXECUTED
* :addons:languages:nias:apk:extractProguardFiles state: EXECUTED
* :addons:languages:nias:pack:bundleLibResRelease state: SKIPPED
* :addons:languages:nias:pack:bundleLibRuntimeToJarRelease state: SKIPPED
* :addons:languages:nias:pack:createFullJarRelease state: SKIPPED
* :addons:languages:nias:pack:writeReleaseLintModelMetadata state: SKIPPED
* :addons:languages:nias:apk:lintVitalAnalyzeRelease state: EXECUTED
* :addons:languages:nias:apk:lintVitalReportRelease state: EXECUTED
* :addons:languages:nias:apk:lintVitalRelease state: EXECUTED
* :api:copyReleaseJniLibsProjectOnly state: SKIPPED
* :addons:base:copyReleaseJniLibsProjectOnly state: SKIPPED
* :addons:languages:nias:apk:mergeReleaseJniLibFolders state: SKIPPED
* :addons:languages:nias:pack:copyReleaseJniLibsProjectOnly state: SKIPPED
* :addons:languages:nias:apk:mergeReleaseNativeLibs state: SKIPPED
* :addons:languages:nias:apk:stripReleaseDebugSymbols state: SKIPPED
* :addons:languages:nias:apk:extractReleaseNativeSymbolTables state: SKIPPED
* :addons:languages:nias:apk:mergeReleaseNativeDebugMetadata state: SKIPPED
* :addons:languages:nias:apk:checkReleaseDuplicateClasses state: SKIPPED
* :addons:languages:nias:apk:dexBuilderRelease state: EXECUTED
* :addons:languages:nias:apk:desugarReleaseFileDependencies state: EXECUTED
* :addons:languages:nias:apk:mergeExtDexRelease state: SKIPPED
* :addons:languages:nias:apk:mergeDexRelease state: SKIPPED
* :addons:languages:nias:apk:mergeReleaseArtProfile state: EXECUTED
* :addons:languages:nias:apk:compileReleaseArtProfile state: SKIPPED
* :addons:languages:nias:apk:mergeReleaseShaders state: SKIPPED
* :addons:languages:nias:apk:compileReleaseShaders state: SKIPPED
* :addons:languages:nias:apk:generateReleaseAssets state: SKIPPED
* :addons:languages:nias:apk:mergeReleaseAssets state: SKIPPED
* :addons:languages:nias:apk:compressReleaseAssets state: SKIPPED
* :addons:languages:nias:apk:processReleaseJavaRes state: SKIPPED
* :addons:languages:nias:apk:mergeReleaseJavaResource state: EXECUTED
* :addons:languages:nias:apk:optimizeReleaseResources state: EXECUTED
* :addons:languages:nias:apk:collectReleaseDependencies state: EXECUTED
* :addons:languages:nias:apk:sdkReleaseDependencyData state: EXECUTED
* :addons:languages:nias:apk:writeReleaseAppMetadata state: SKIPPED
* :addons:languages:nias:apk:writeReleaseSigningConfigVersions state: SKIPPED
* :addons:languages:nias:apk:packageRelease state: EXECUTED
* :addons:languages:nias:apk:createReleaseApkListingFileRedirect state: EXECUTED
* :addons:languages:nias:apk:assembleRelease state: EXECUTED
* :generateFdroidYamls state: EXECUTED
* :addons:languages:nias:apk:assemble state: EXECUTED

**Reply by sslaia** [reply]
This is what I see in the console. As you can see below the VersionName is 4.1.-315, which should be 4.0.1415 (version code 2927/337)

....
> Configure project :addons:languages:urdu:apk
Could not find '/tmp/anysoftkeyboard.keystore' file. Release APK will not be signed.
Project :addons:languages:urdu:apk is NOT marked for publishing
Building :addons:languages:urdu:apk 4.1.-315 (or 4.1.-315), code 2927.

> Configure project :addons:quicktexts:bbcodes:apk
Could not find '/tmp/anysoftkeyboard.keystore' file. Release APK will not be signed.
Building :addons:quicktexts:bbcodes:apk 4.1.-315 (or 4.1.-315), code 2927.

> Configure project :addons:themes:classic_pc:apk
Could not find '/tmp/anysoftkeyboard.keystore' file. Release APK will not be signed.
Building :addons:themes:classic_pc:apk 4.1.-315 (or 4.1.-315), code 2927.

> Configure project :addons:themes:ics:apk
Could not find '/tmp/anysoftkeyboard.keystore' file. Release APK will not be signed.
Building :addons:themes:ics:apk 4.1.-315 (or 4.1.-315), code 2927.

> Configure project :addons:themes:israel64:apk
Could not find '/tmp/anysoftkeyboard.keystore' file. Release APK will not be signed.
Building :addons:themes:israel64:apk 4.1.-315 (or 4.1.-315), code 2927.

> Configure project :addons:themes:three_d:apk
Could not find '/tmp/anysoftkeyboard.keystore' file. Release APK will not be signed.
Project :addons:themes:three_d:apk is NOT marked for publishing
Building :addons:themes:three_d:apk 4.1.-315 (or 4.1.-315), code 2927.
Found prebuilt word-list folder for :addons:languages:nias:pack with 3 files.

> Task :addons:languages:nias:pack:makeDictionary
Reading words from input /home/Projects/Java/AnySoftKeyboard/addons/languages/nias/pack/build/dictionary/words_merged.xml
Will store output files under /home/Projects/Java/AnySoftKeyboard/addons/languages/nias/pack/src/main/res/raw. Created raw folder? true
Deleting previous versions...
Nodes = 32994
Dict Size = 149823
Writing to dict file /home/Projects/Java/AnySoftKeyboard/addons/languages/nias/pack/src/main/res/raw/nias_words_1.dict
Done. Wrote 1 files.
Done.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.4.2/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 20s
217 actionable tasks: 71 executed, 139 from cache, 7 up-to-dat

**Reply by sslaia** [reply]
_this should not happen, can you paste the gradle log while building?_

Hi @menny I have found the line where the number is not correct converted. It's in the file **build.grade** in the **addons** folder line 11:

>     patchOffset = -1512-1730

This line used to be with only one number **-1512**, but now there's also **-1730**. I didn't know this change and if I remove the second number (-1730) the addons compiles correctly.

What should I do? I suppose, you added the second number surely for a reason.

**Reply by menny** [reply]
What arguments do you provide in the build command line? Any build-number related environment variables are set up in your environment?

**Reply by sslaia** [reply]
Thank you for looking into this. The first argument is
>  export BUILD_COUNT_FOR_VERSION=337

and then

> ./gradlew -PwithAutoVersioning :addons:languages:nias:apk:assembleDebug

**Comment by menny** [comment]
Ah! Okay, now I understand. So, the `BUILD_COUNT_FOR_VERSION=337` is a
variable that is set by our build system - every new build increments this
value by one.
Are you trying to produce a versioned APK for releasing? Or just build for
development?

**Comment by sslaia** [comment]
I'd like to build the release version of everything is ok. The existing one
has the variable 336, so the next one will be 337.

**Comment by sslaia** [comment]
I found a solution: the value for BUILD_COUNT_FOR_VERSION needs to be bumped up. Any numbers greater than 653 works.

I don't know how to explain it. The latest update I made had the value 336. However setting the next value to 337 would produce wrong versionName (with minus in it).

Thank you @menny for your patience with me.

Edit: spelling.

**Reply by menny** [reply]
yes, that makes sense. The build-counter should be incremented with each release. When we do a minor version update (for example, 4.0->4.1) we also update the offset value, I'm guessing this is why you needed to increment your build-counter by so much.
